src/main/java/com/qubit/solution/fenixedu/integration/cgd/domain/logs/CgdCommunicationLog.java

```diff
@@ -0,0 +1,45 @@
+package com.qubit.solution.fenixedu.integration.cgd.domain.logs;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.fenixedu.academic.domain.Person;
+import org.fenixedu.academic.domain.student.Registration;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.joda.time.DateTime;
+
+public class CgdCommunicationLog extends CgdCommunicationLog_Base {
+
+    public CgdCommunicationLog(Registration registration, boolean requestCard, boolean success, Person sender) {
+        setRegistration(registration);
+        setRequestCard(requestCard);
+        setSuccess(success);
+        setSender(sender);
+        setSendDate(DateTime.now());
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    public static Collection<CgdCommunicationLog> findLogsByStudent(Person person) {
+        return Bennu.getInstance().getCgdComunicationLogsSet().stream()
+                .filter(log -> log.getRegistration().getPerson().equals(person))
```

Comment:
@miguel-morais we're comparing domain objects so we can use == operator since domain objects are ensured to be unique. Also it shouldn't happen but we've already seen some registrations without a person (yes...data integrity issues, but still..)
So I would change this into `log.getRegistration().getPerson() == person`